common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -122,6 +131,37 @@ protected void append(LoggingEvent event) {
             System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
             isTruststore = true;
         }
+        if (dataPublisher == null) {
+            Pattern pattern = Pattern.compile("(tcp):\\/\\/([a-zA-Z0-9]+):([0-9]+)");
+            Matcher matcher = pattern.matcher(url);
+            if (matcher.find() && authURLSet == null) {
+                authURLSet = "ssl://" + matcher.group(2).toString() + ":" + (
+                        Integer.parseInt(matcher.group(3).toString()) + 100);
+            }
+            try {
+                dataPublisher = new DataPublisher("Thrift", url, authURLSet, userName, password);
```

Comment:
Hi Ruwan,

This scenario is handle when we adding the secure value feature, as per update version each log event before publishing the log event it is check whether data publisher in initialize unless it call the publisherInitializer() method and do the initialization.

Please find the relevant commit on : https://github.com/wso2/shared-analytics/commit/7fe7d54af26dbfef4e1c1965e95c0898fe5406bb

Thanks,
SajithD